### ISSUE TITLE
Binary type ( bool ) not handled correctly in queries

### ISSUE THREAD

**Issue by eczema** [netfishers-onl/Netshot#27]
Bool type use = '1' in queries...

Postgresql use 'is true' .... 

So depending drivers, the query must be adpted.... 

quick and dirty fix to test postgresql
````
                public FinderCriteria buildHqlString(String itemPrefix) {
                        FinderCriteria criteria = super.buildHqlString(itemPrefix);
                        //criteria.where = this.buildWhere("assumption", "=", "1");
                        criteria.where = this.buildWhere("assumption", "is", "true");
                        return criteria;
                }
````

seems to be compliant with mysql as well 

using : ` select * from device_attribute where assumption is true ;`  

give same result as

 ` select * from device_attribute where assumption = 1 ;`  

Do you agree if we use is true instead of = 1 ??? 

E.

**Comment by eczema** [comment]
Hold this , seems some issue on this test... need to investigate

**Comment by eczema** [comment]
the problem is the  'NOT' token not handled correctly with bool....

**Comment by eczema** [comment]
Tested sucessfully this way to handle true/false..; What do you think ? 

```
                /* (non-Javadoc)
                 * @see onl.netfishers.netshot.device.Finder.Expression#buildHqlString(java.lang.String)
                 */
                public FinderCriteria buildHqlString(String itemPrefix) {
                        FinderCriteria criteria = this.child.buildHqlString(itemPrefix + "_0");
                        if (  criteria.where.indexOf("is true") > 0 ) {
                                criteria.where = criteria.where.replaceAll("is true", "is false");
                        } else {
                                criteria.where = "not (" + criteria.where + ")";
                        }
                        return criteria;
                }
```

**Comment by SCadilhac** [comment]
Fixed by 4c52f145f1ab8e2102c9f23f7fe78744d942d81d.
